src/py21cmfast/src/HaloBox.c

```diff
@@ -329,32 +477,219 @@ int get_box_averages(double redshift, double norm_esc, double alpha_esc, double
+                          struct PerturbHaloField *halos, struct HaloBoxConstants *consts, struct HaloBox *grids, struct HaloProperties *averages){
+    double redshift = consts->redshift;
+    //averages
+    double hm_avg=0.,sm_avg=0.,sfr_avg=0.;
+    double sm_avg_mini=0.,sfr_avg_mini=0.;
+    double M_turn_a_avg=0.,M_turn_m_avg=0.,M_turn_r_avg=0.;
+    double n_ion_avg=0., wsfr_avg=0., xray_avg=0.;
+    //counts
+    unsigned long long int total_n_halos, n_halos_cut=0.;
+
+    double cell_volume = VOLUME / HII_TOT_NUM_PIXELS;
+    #pragma omp parallel num_threads(user_params_stoc->N_THREADS)
+    {
+        int x,y,z;
+        unsigned long long int i_halo,i_cell;
+        double m,nion,sfr,wsfr,sfr_mini,stars_mini,stars,xray;
+        double J21_val, Gamma12_val, zre_val;
+
+        double curr_vcb = consts->vcb_norel;
+        double M_turn_m = consts->mturn_m_nofb;
+        double M_turn_a = consts->mturn_a_nofb;
+        double M_turn_r;
+
+        float in_props[3];
+        float out_props[7];
+
+    #pragma omp for reduction(+:hm_avg,sm_avg,sm_avg_mini,sfr_avg,sfr_avg_mini,M_turn_a_avg,M_turn_m_avg,M_turn_r_avg,n_halos_cut)
+        for(i_halo=0; i_halo<halos->n_halos; i_halo++){
+            m = halos->halo_masses[i_halo];
+            //It is sometimes useful to make cuts to the halo catalogues before gridding.
+            //  We implement this in a simple way, if the user sets a halo's mass to zero we skip it
+            if(m == 0.){
+                n_halos_cut++;
+                continue;
+            }
+            x = halos->halo_coords[0+3*i_halo]; //NOTE:PerturbHaloField is on HII_DIM, HaloField is on DIM
+            y = halos->halo_coords[1+3*i_halo];
+            z = halos->halo_coords[2+3*i_halo];
+
+            //set values before reionisation feedback
+            //NOTE: I could easily apply reionization feedback without minihalos but this was not done previously
+            if(flag_options_stoc->USE_MINI_HALOS){
+                if(!flag_options_stoc->FIX_VCB_AVG && user_params_stoc->USE_RELATIVE_VELOCITIES)
+                    curr_vcb = ini_boxes->lowres_vcb[HII_R_INDEX(x,y,z)];
+
+                J21_val = redshift < global_params.Z_HEAT_MAX ? previous_spin_temp->J_21_LW_box[HII_R_INDEX(x,y,z)] : 0.;
+                Gamma12_val = redshift < global_params.Z_HEAT_MAX ? previous_ionize_box->Gamma12_box[HII_R_INDEX(x, y, z)] : 0.;
+                zre_val = redshift < global_params.Z_HEAT_MAX ? previous_ionize_box->z_re_box[HII_R_INDEX(x, y, z)] : 0.;
+                M_turn_m = lyman_werner_threshold(redshift, J21_val, curr_vcb, astro_params_stoc);
+                M_turn_r = reionization_feedback(redshift, Gamma12_val, zre_val);
+                M_turn_a = consts->mturn_a_nofb < M_turn_r ? M_turn_r : consts->mturn_a_nofb; //since ACG turnover not reassigned
+                if(M_turn_m < M_turn_r)M_turn_m = M_turn_r;
```

Comment:
why use the `if` here instead of the `?`?

Reply:
Actually, the set of three lines above comparing `redshift` to `Z_HEAT_MAX` would probably be easier to read if you just had:

```c
J21_val=0.; Gamma12_val =0.; zre_val = 0;
if(redshift < global_params.Z_HEAT_MAX){
    J21_val = J_21_LW_box[HII_R_INDEX(x,y,z)];
   etc.
```

Reply:
In general, I use ternaries `a = X ? b : c`  when neither `b==a` or `c==a`, when one of the cases leaves the variable alone I use `if`.

I'll change both sets to initialisation + `if` for readability.

Reply:
That sounds like the right approach